### PR TITLE
Built out User testing

### LINKED ISSUE #43: Build out unit testing for /users endpoints
**Is your feature request related to a problem? Please describe.**

We want to ensure that our CRUD functionality in all edge cases is working. So we'll use unit and integration testing with Mocha as our test runner, Chai as our assertion library, and Sinon for spies/mocks/stubs.

**Describe the solution you'd like**

Every endpoint for `/users` should be tested against different edge cases to make sure it's doing exactly what we would like it to do. These tests will become part of our continuous integration suite, meaning that they'll run after every commit to our repo and serve as not only proof that what we're building does what it's intended, but also prevents regressions in functionality.

**Additional context**
Add details, any other context or screenshots about the feature request here.

### PR DESCRIPTION
**Pull request by gwang0136**
:construction: In development

## Description

Built out unit testing to test user endpoints.

Fixes #43 

## Todos
Test more edge cases.

### CODE REVIEW

File: backend/test/user.test.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+const stubs = require("./auth_stubs").stubAllAuth();
+const app = require("../app.js");
+const User = require("../models/User");
+
+beforeEach(() => User.remove({}));
+
+const sampleUserInfo = {
+  firstName: "alan",
+  lastName: "fang",
+  email: "analfang@illinois.edu",
```
Comment (alan-fang) [inline backend/test/user.test.js]: 😶
